kafka-client/src/main/java/dev/responsive/kafka/config/ResponsiveConfig.java

```diff
@@ -234,29 +234,27 @@ public class ResponsiveConfig extends AbstractConfig {
           REMOTE_TABLE_CHECK_INTERVAL_MS_DOC
       );
 
-  private static class NonEmptyPassword implements Validator {
-    private final String passwordConfig;
-
-    NonEmptyPassword(final String passwordConfig) {
-      this.passwordConfig = passwordConfig;
-    }
-
-    @Override
-    public void ensureValid(String name, Object o) {
-      Password p = (Password) o;
-      if (p != null && p.value().isEmpty()) {
-        throw new ConfigException(name, o, passwordConfig + " must be non-empty");
-      }
-    }
+  /**
+   * This should generally be used only once per app to avoid excessive logging. Prefer
+   * {@link #quietConfig(Map)} whenever you need to construct a {@code ResponsiveConfig}
+   * mid-application, for example during state store init.
+   */
+  public static ResponsiveConfig loggedConfig(final Map<?, ?> originals) {
+    return new ResponsiveConfig(originals, true);
+  }
 
-    @Override
-    public String toString() {
-      return "non-empty password";
-    }
+  /**
+   * This should generally be used over the {@link #loggedConfig(Map)} override in all
+   * cases outside the initial {@link dev.responsive.kafka.api.ResponsiveKafkaStreams#create}.
+   * Prefer this API whenever you need to construct a {@code ResponsiveConfig} mid-application,
+   * for example during state store init.
+   */
+  public static ResponsiveConfig quietConfig(final Map<?, ?> originals) {
```

Comment:
nit: maybe just call this `config` so that it's perceived as the default to use

Reply:
good call